pkg/storage/bloom/v1/versioned_builder.go

```diff
@@ -0,0 +1,218 @@
+package v1
+
+import "github.com/pkg/errors"
+
+/*
+Each binary format (version) has it's own builder. This provides type-safe way to build the binary format
+while allowing reuse of underlying logic. As an example, the V2Builder will prevent encoding v1 series (only 1 bloom per series)
+as it only provides methods that are v2 compatible. The opposite is also true.
+
+Builders provide the following methods:
```

Comment:
nit: define Interface and add docs for each method. laso add checks to make sure both builders satisfy the interface at compile time.

Reply:
nvm. I see the signature of some of the methods is slightly different.